File: 69-sqrtx/sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left=0
        right=x
        while left<=right:
            mid=(left+right)//2
            if mid*mid==x:
                return mid
            elif mid*mid < x:
                left = mid + 1
            else:
                right = mid - 1
        return left-1

        '''
        for i in range(x+1):
            if i*i==x:
                return i
            if i*i>x:
                return i-1
                '''

File: 69-sqrtx/sqrtx.js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    for (let i = 1; i<=x;i++){
        if (i*i===x){
            return i
        }
        else if (i*i>x){
            return i-1
        }
    }
    return 0
};